sqlmesh/core/selector.py

```diff
@@ -12,25 +11,21 @@
 from sqlmesh.utils.dag import DAG
 from sqlmesh.utils.errors import SQLMeshError
 
+if t.TYPE_CHECKING:
+    from sqlmesh.core.context import Context
+
 
 class Selector:
     def __init__(
         self,
         state_reader: StateReader,
         models: UniqueKeyDict[str, Model],
-        context_path: Path = Path("."),
+        context: Context,
```

Comment:
@izeigerman i know this will upset you, but get the right config per model is not trivial which is needed for update_model_schemas

Reply:
I'm absolutely against this :)

Reply:
We should then refactor how we get a config per model.

Reply:
Made a better suggestion in the other thread